src/components/Flyout/useContainScroll.ts

```diff
@@ -0,0 +1,46 @@
+import { MutableRefObject, useEffect } from "react";
+
+export const useContainScroll = (overlayRef: MutableRefObject<HTMLDivElement | null>, { isDisabled = false }) => {
+    useEffect(() => {
+        const overlay = overlayRef.current;
+
+        const scrollHandler = (event: WheelEvent) => {
+            const prevent = () => {
+                event.stopPropagation();
+                event.preventDefault();
+            };
+
+            if (overlay !== null) {
+                const deltaY = event.deltaY;
+                const up = deltaY < 0;
+                const scrollTop = overlay.scrollTop;
+                const scrollHeight = overlay.scrollHeight;
+                const height = overlay.clientHeight;
```

Comment:
You could destruct like so:
```suggestion
                const { scrollTop, scrollHeight, clientHeight } = overlay;
```

Reply:
done!